benchexec/result.py

```
@@ -314,16 +332,93 @@ def get_result_classification(result):
+            is_unknown = True
+    if not is_correct:
+        return CATEGORY_WRONG
+    if is_error:
+        return CATEGORY_ERROR
+    if is_unknown:
+        return CATEGORY_UNKNOWN
+    return CATEGORY_CORRECT
+
+
+def _score_for_multiproperty(actual_statuses, ideal_statuses, result):
+    assert not (bool(ideal_statuses.keys() - actual_statuses.keys()))
+    score = 0
+    for property in ideal_statuses.keys():
+        actual_status = str(actual_statuses[property]).lower()
+        ideal_status = str(ideal_statuses[property]).lower()
+
+        if ideal_status == RESULT_CLASS_TRUE:
+            if not result:
+                score += _SCORE_CORRECT_TRUE
+            elif actual_status == RESULT_CLASS_TRUE:
+                score += _SCORE_CORRECT_TRUE
+            elif actual_status == RESULT_CLASS_FALSE:
+                score += _SCORE_WRONG_FALSE
+            else:
+                score += _SCORE_UNKNOWN
+        elif ideal_status == RESULT_CLASS_FALSE:
+            if not result:
+                score += _SCORE_CORRECT_FALSE
+            elif actual_status == RESULT_CLASS_TRUE:
+                score += _SCORE_WRONG_TRUE
+            elif actual_status == RESULT_CLASS_FALSE:
+                score += _SCORE_CORRECT_FALSE
+            else:
+                score += _SCORE_UNKNOWN
+        else:
+            score += _SCORE_UNKNOWN
+    return score
+
+
+def _get_yaml_ideal_statuses(filename):
+    # Multiple properties are being checked.
+    # We need to get results from specific YAML file.
+    ideal_statuses = {}
+    specification_file = filename + _YAML_EXTENSION
+    with open(specification_file, 'r') as f:
+        try:
+            content = yaml.load(f)
+            for (property, status) in content.get("correct results").items():
+                ideal_statuses[property] = status
+        except yaml.YAMLError as exc:
+            # Wrong format, ideal statuses were failed to obtained.
+            pass
```

Swallowing errors silently is not good.